pandas/core/missing.py

```diff
@@ -621,3 +622,24 @@ def fill_zeros(result, x, y, name, fill):
             result = result.reshape(shape)
 
     return result
+
+
+def maybe_fill(obj, value, errors):
```

Comment:
This is a non-idempotent function that either raises or returns a flag depending on the mode (?).

For the `fillna` use-case alone one way to avoid this combination would be to add an `if-else` to the method body, and split this logic into two separate methods in `missing` (so e.g. if mode is errors do `validate_fill_value`, otherwise do `continue_to_fill`).

This would be fine for `fillna` alone. But, I'm trying to keep an eye out also for extending this raise/ignore/coerce stuff to other methods as well in some later PRs (#15533), and obviously you can't go about inserting this same `if-else` block everywhere all willy-nilly.